### ISSUE TITLE
Latest QGIS fails to load on Linux Mint

### ISSUE THREAD

**Issue by kaptainkory** [qgis/QGIS#37343]
**Describe the bug**
Latest QGIS (1:3.15.0+git20200620+e4718d7471+28bionic) fails to load in Linux Mint.  This just happened after latest update.

**How to Reproduce**
Run "qgis" command from terminal.

![Screenshot from 2020-06-22 14-08-05](https://user-images.githubusercontent.com/59102670/85325789-d4b34f00-b491-11ea-8652-a72ecd9915c1.png)

**Comment by gioman** [comment]
1:3.15.0+git20200620+e4718d7471+28bionic works ok on Ubuntu 18.04 installed from qgis.org repos.

**Comment by kaptainkory** [comment]
I'm using this repo: deb https://qgis.org/ubuntu-nightly bionic main

This might possibly offer a hint at the issue?

https://gis.stackexchange.com/questions/332793/python-importerror-with-libgdal-undefined-symbol-opj-codec-set-threads

I uninstalled QGIS by right-clicking icon and "uninstall" from the Linux Mint (cinnamon) menu.

Changed repo to deb https://qgis.org/ubuntu-nightly-release bionic main

Tried to reinstall, but ran into dependency conflicts.  Went into Synaptic and removed a ton of QGIS libraries that got left behind (unaffected by uninstall).

![Screenshot from 2020-06-22 19-15-01](https://user-images.githubusercontent.com/59102670/85346967-bb73c800-b4bc-11ea-8c8a-5fc0f1a26aea.png)

Ran "sudo apt-get install qgis qgis-plugin-grass" from terminal.  Still no good.

`/usr/lib/qgis/crssync: symbol lookup error: /usr/lib/libgdal.so.20: undefined symbol: sqlite3_column_table_name
dpkg: error processing package qgis-providers (--configure):
 installed qgis-providers package post-installation script subprocess returned error exit status 127
dpkg: dependency problems prevent configuration of qgis:
 qgis depends on qgis-providers (= 1:3.12.2+git20200515+9c0093c7ac+28bionic); however:
  Package qgis-providers is not configured yet.

dpkg: error processing package qgis (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of python3-qgis:
 python3-qgis depends on qgis-providers (= 1:3.12.2+git20200515+9c0093c7ac+28bionic); however:
  Package qgis-providers is not configured yet.

dpkg: error processing package python3-qgis (--configure):
 dependency problems - leaving unconfigured
Setting up libqgis-customwidgets (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
dpkg: dependency problems prevent configuration of qgis-plugin-grass:
 qgis-plugin-grass depends on qgis (= 1:3.12.2+git20200515+9c0093c7ac+28bionic); however:
  Package qgis is not configured yet.

dpkg: error processing package qgis-plugin-grass (--configure):
 dependency problems - leaving unconfigured
Setting up libqgisNo apport report written because the error message indicates its a followup error from a previous failure.
                                            No apport report written because the error message indicates its a followup error from a previous failure.
                                                                      No apport report written because MaxReports is reached already
                                                    grass7-3.12.2 (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
Setting up qgis-provider-grass (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
Setting up python3-qgis-common (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
Processing triggers for desktop-file-utils (0.23+linuxmint8) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
Processing triggers for shared-mime-info (1.9-2) ...
Processing triggers for gnome-menus (3.13.3-11ubuntu1.1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Errors were encountered while processing:
 qgis-providers
 qgis
 python3-qgis
 qgis-plugin-grass
E: Sub-process /usr/bin/dpkg returned an error code (1)`

I uninstalled everything that matched to "gdal" in Synaptic.

![Screenshot from 2020-06-22 19-25-20](https://user-images.githubusercontent.com/59102670/85347460-3b4e6200-b4be-11ea-8dc0-aca7ce3517e2.png)

Ran "sudo apt-get install qgis qgis-plugin-grass" again from terminal.  Still no good.  Same hang-up.

`/usr/lib/qgis/crssync: symbol lookup error: /usr/lib/libgdal.so.20: undefined symbol: sqlite3_column_table_name
dpkg: error processing package qgis-providers (--configure):
 installed qgis-providers package post-installation script subprocess returned error exit status 127
dpkg: dependency problems prevent configuration of qgis:
 qgis depends on qgis-providers (= 1:3.12.2+git20200515+9c0093c7ac+28bionic); however:
  Package qgis-providers is not configured yet.

dpkg: error processing package qgis (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                          dpkg: dependency problems prevent configuration of python3-qgis:
 python3-qgis depends on qgis-providers (= 1:3.12.2+git20200515+9c0093c7ac+28bionic); however:
  Package qgis-providers is not configured yet.

dpkg: error processing package python3-qgis (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                          Setting up libqgis-customwidgets (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
dpkg: dependency problems prevent configuration of qgis-plugin-grass:
 qgis-plugin-grass depends on qgis (= 1:3.12.2+git20200515+9c0093c7ac+28bionic); however:
  Package qgis is not configured yet.

dpkg: error processing package qgis-plugin-grass (--configure):
 dependency problems - leaving unconfigured
Setting up libqgisgrass7-3.12.2 (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
No apport report written because MaxReports is reached already
                                                              Setting up qgis-provider-grass (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
Setting up python3-qgis-common (1:3.12.2+git20200515+9c0093c7ac+28bionic) ...
Processing triggers for desktop-file-utils (0.23+linuxmint8) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
Processing triggers for shared-mime-info (1.9-2) ...
Processing triggers for gnome-menus (3.13.3-11ubuntu1.1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Errors were encountered while processing:
 qgis-providers
 qgis
 python3-qgis
 qgis-plugin-grass
E: Sub-process /usr/bin/dpkg returned an error code (1)
`

**Comment by gioman** [comment]
@kaptainkory there are no issues (at least here) on Bionic installing from qgis.org repos. You probably have some mix up/local issues. Please try purge/remove any qgis/gdal/proj packages, then check your sources (to check you are not mixing up repos, i.e. if you are using https://qgis.org/ubuntu-nightly you should not use also the ubuntugis ppa), update sources and install again.

**Comment by kaptainkory** [comment]
I did all that.  I can't figure it out.  Running "gdalinfo" without even trying to start QGIS gives me the same error.  I know this is probably not a bug specific to QGIS, but I'm completely stuck.

![Screenshot from 2020-06-23 12-40-12](https://user-images.githubusercontent.com/59102670/85436455-b5c6c280-b54e-11ea-8874-8d15141ac740.png)

It appears this is not a new or isolated issue with GDAL/QGIS, however.

http://osgeo-org.1560.x6.nabble.com/QGIS-doesn-t-start-anymore-undefined-symbol-pj-ctx-alloc-td5016501.html

**Comment by gioman** [comment]
> I know this is probably not a bug specific to QGIS,

I agree, I don't see any issue installing/updating/upgrading on my Ubuntu 18.04 machine.



> It appears this is not a new or isolated issue with GDAL/QGIS, however.
> 
> http://osgeo-org.1560.x6.nabble.com/QGIS-doesn-t-start-anymore-undefined-symbol-pj-ctx-alloc-td5016501.html

8 years old... and an isolated case, and probably also a local mix up.



> Or with sqlite?
> 
> https://askubuntu.com/questions/1097177/installing-qgis-in-ubuntu-18-04-error-with-sqlite-tables-can-no-longer-use-apt

also likely a local mix up.


Closing?

**Comment by kaptainkory** [comment]
Close it.  I'm only offering the following for others in the future:

Somewhere along the way I had compiled and installed sqlite from source (because my system didn't have a version recent enough to offer windowing, which I wanted to test out).  GDAL doesn't work unless sqlite is compiled with a nonstandard switch:

"This requires GDAL to be compiled with the SQLITE_HAS_COLUMN_METADATA option and SQLite3 with the SQLITE_ENABLE_COLUMN_METADATA option."

https://gdal.org/drivers/vector/gpkg.html

...and QGIS doesn't work unless GDAL does.

I couldn't get sqlite to recompile with the switch, but I was able to uninstall with "make uninstall" from the configuration directory.

Aside from cleaning up the remaining mess of repos and packages, now works.
